app/controllers/creation_controller.rb

```diff
@@ -98,8 +98,9 @@ def extract_error_messages_from_api_exception(api_message)
     api_errors_hash.key?('general') ? api_errors_hash['general'] : [api_message]
   end
 
-  def redirect_back_after_error(prefix_message, error_messages)
-    flash_messages = [prefix_message] + error_messages
+  # Use of splat on parameter to cope with both arrays of error messages and single strings
+  def redirect_back_after_error(prefix_message, *error_messages)
+    flash_messages = [prefix_message] + error_messages.flatten
```

Comment:
I think the string is originating from line 36, where we're splatting out the array. If this was hit with an array of multiple errors, it would have been throwing an unknown number of arguments error, but as it was we had one item in the array, so it became a single string.

I'm tempted to suggest:
1) Remove the * from line 36
2) Keep this as a single parameter

This would fix the issue, although would leave is vulnerable if the SS API ever returned a string under 'general', which *shouldn't* happen, but you never know.

If you wanted to proof it agains this, or strings from other resources you could do:

```ruby
flash_messages = [prefix_message] + Array(error_messages)
```

Because:

```ruby
irb(main):001:0> Array('a')
=> ["a"]
irb(main):002:0> Array(['a','b'])
=> ["a", "b"]
```

This discomfort mostly stems from this approach leaving two potential ways to pass in an array of errors:

```ruby
redirect_back_after_error('One way', ['is like this', 'with an array as a single argument'])
redirect_back_after_error('The other way', *['is like this', 'with an array as a single argument'])
# Which is equivalent to
redirect_back_after_error('The other way', 'is like this', 'with an array as a single argument')
```